File: Sample/GameNotice/release/Script/PyScript/Action/action2001.py
"""2001_游戏公告信息【完成】"""
import ReferenceLib
from action import *
from ZyGames.Framework.Common import *
from ZyGames.Framework.Cache.Generic import *
from ZyGames.Framework.Game.Cache import *
from ZyGames.Framework.Game.Service import *
from GameNotice.Model import *


class UrlParam(HttpParam):
    def __init__(self):
        HttpParam.__init__(self)
        self.GameType = 0
        self.ServerID = 0
        self.PageIndex = 0
        self.PageSize = 0


class ActionResult(DataResult):
    def __init__(self):
        DataResult.__init__(self)
        self.PageCount = 0
        self.DsItemCollect = None


def getUrlElement(httpGet, parent):
    urlParam = UrlParam()
    if httpGet.Contains("GameType")\
    and httpGet.Contains("ServerID")\
    and httpGet.Contains("PageIndex")\
    and httpGet.Contains("PageSize"):
        urlParam.GameType = httpGet.GetIntValue("GameType")
        urlParam.ServerID = httpGet.GetIntValue("ServerID")
        urlParam.PageIndex = httpGet.GetIntValue("PageIndex")
        urlParam.PageSize = httpGet.GetIntValue("PageSize")
    else:
        urlParam.Result = False

    return urlParam

def takeAction(urlParam, parent):
    actionResult = ActionResult()
    noticeList = ShareCacheStruct[Notice]().FindAll()
    result = MathUtils.GetPaging[Notice](noticeList, urlParam.PageIndex, urlParam.PageSize)
    if result:
        actionResult.dsItemCollect = result[0]
        actionResult.PageCount = result[1]
    return actionResult

def buildPacket(writer, urlParam, actionResult):
    writer.PushIntoStack(actionResult.PageCount)
    writer.PushIntoStack(len(actionResult.dsItemCollect))
    for info in actionResult.dsItemCollect:
        dsItem = DataStruct()
        dsItem.PushIntoStack(info.Title)
        dsItem.PushIntoStack(info.Content)
        dsItem.PushIntoStack(info.CreateDate.ToString("yyyy-MM-dd HH:mm:ss"))
        writer.PushIntoStack(dsItem)


    return True

File: Sample/GameNotice/src/GameNotice/Script/LuaScript/Action/Action2001.lua
Action2001 ={}

function Action2001:getUrlElement(httpGet, parent)
	local urlParam = {}
	urlParam.Result = true
	urlParam.GameType = ReadNumberParam(httpGet, "GameType")
    urlParam.ServerID = ReadNumberParam(httpGet, "ServerID")
    urlParam.PageIndex = ReadNumberParam(httpGet, "PageIndex")
    urlParam.PageSize = ReadNumberParam(httpGet, "PageSize")
    return urlParam
end

function Action2001:takeAction(urlParam, parent)
	CPrint("lua print test...")
	local actionResult = {}
	actionResult.PageCount = 0
	actionResult.ItemCount = 0
	actionResult.dsItemCollect = nil
	actionResult.Result = true
	local result =  GetNoticeCache(urlParam.PageIndex, urlParam.PageSize)
	if result == nil then
		actionResult.Result = false
		return actionResult
	end
	actionResult.PageCount = result[0]
	actionResult.dsItemCollect = result[1]
	actionResult.ItemCount = actionResult.dsItemCollect.Count
    return actionResult
end


function Action2001:buildPacket(writer, urlParam, actionResult)
	PushIntoStack(writer, actionResult.PageCount)
    PushIntoStack(writer, actionResult.ItemCount)
	local len = actionResult.ItemCount
    for i=1,len,1 do
		local info = actionResult.dsItemCollect[i-1]
        dsItem = CreateDataStruct()
        PushIntoStack(dsItem, info.Title)
        PushIntoStack(dsItem, info.Content)
        PushIntoStack(dsItem, FormatDateString(info.CreateDate, "yyyy-MM-dd HH:mm:ss"))
        PushIntoStack(writer, dsItem)
	end
    return true
end

